# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/PContent.ipdl
# Commit: 40fcc0cdc593
# Full Hash: 40fcc0cdc593fa337273631de435a2901ec881f6
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2024-02-07 04:17:40
# Regressor Bug: 1878286
# File Overlap Count: 2
# Description:
#   Bug 1878286 - Eliminate usage of fontlist::Pointer in IPDL. r=gfx-reviewers,lsalzman
#   
#   No change in behavior.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D200564
# ==============================================================================

diff -r c48ca61bc925 -r 40fcc0cdc593 dom/ipc/PContent.ipdl
--- a/dom/ipc/PContent.ipdl	Tue Feb 06 10:15:29 2024 +0000
+++ b/dom/ipc/PContent.ipdl	Tue Feb 06 10:41:50 2024 +0000
@@ -132,7 +132,6 @@
 using mozilla::dom::MaybeDiscardedWindowContext from "mozilla/dom/WindowContext.h";
 using mozilla::dom::WindowContextTransaction from "mozilla/dom/WindowContext.h";
 [MoveOnly] using base::SharedMemoryHandle from "base/shared_memory.h";
-using mozilla::fontlist::Pointer from "SharedFontList.h";
 using gfxSparseBitSet from "gfxFontUtils.h";
 using FontVisibility from "gfxFontEntry.h";
 using mozilla::dom::MediaControlAction from "mozilla/dom/MediaControlKeySource.h";
@@ -1441,24 +1440,27 @@
      * @param aGeneration
      *   Font-list generation, so requests relating to an obsolete list can be
      *   ignored (see comments for GetFontListShmBlock).
-     * @param aFacePtr
-     *   Font-list shared-memory "pointer" to the Face record to be updated.
-     *   A Pointer is a record of a shared-memory block index and an offset
-     *   within that block, which each process that maps the block can convert
-     *   into a real pointer in its address space.
+     * @param aFamilyIndex
+     *   Index of the font family in the font list (see aAlias for which list).
+     * @param aAlias
+     *   Whether aFamilyIndex refers to the Families() or AliasFamilies() list.
+     * @param aFaceIndex
+     *   Index of the face within the family's Faces() list.
      * @param aMap
      *   The character coverage map of the face. (This will be stored as a
      *   SharedBitSet record within the shared font list, and the Face record
      *   will be updated to reference it.)
      */
-    async SetCharacterMap(uint32_t aGeneration, Pointer aFacePtr, gfxSparseBitSet aMap);
+    async SetCharacterMap(uint32_t aGeneration, uint32_t aFamilyIndex, bool aAlias,
+                          uint32_t aFaceIndex, gfxSparseBitSet aMap);
 
     /**
      * Ask the parent to set up the merged charmap for a family, to accelerate
      * future fallback searches.
-     * aFamilyPtr may refer to an element of either the Families() or AliasFamilies().
+     * aFamilyIndex may refer to an element in either Families() or AliasFamilies(),
+     * with aAlias determining which.
      */
-    async SetupFamilyCharMap(uint32_t aGeneration, Pointer aFamilyPtr);
+    async SetupFamilyCharMap(uint32_t aGeneration, uint32_t aFamilyIndex, bool aAlias);
 
     /**
      * Ask the parent to try and complete the InitOtherFamilyNames task, because